src/ngraph/graph_util.cpp

```diff
@@ -435,15 +462,15 @@ bool ngraph::is_one(std::shared_ptr<Node> reduce_constant)
     return result_bool;
 }
 
-bool ngraph::is_used(std::shared_ptr<ngraph::Node> node)
+bool ngraph::is_used(Node* node)
```

Comment:
is there a reason why we are switching to raw pointers? It adds more clutter and makes code less readable.

Reply:
shared_ptr is in general slower since it will need to increment a ref count atomically. I prefer using it only if we need it or if the compiler can eliminate it.